### PR TITLE
NIAD-3137: Add missing test for problem extension.extension url

### PR DESCRIPTION
**Pull request by MartinWheelerMT**
## What

Add missing test to verify that an unspecified problem is produced when a related problem has a problem extension with an extension with an invalid url.

## Why

This functionality was present in the adaptor but was untested

## Type of change

Please delete options that are not relevant.

- [x] Internal change (non-breaking change with no effect on the functionality affecting end users)

## Checklist:

- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterComponentsMapperTest.java [hunk 1]
```diff
@@ -405,6 +405,25 @@ public void When_MappingWithRelatedProblemWithIncorrectProblemExtensionUrl_Expec
             .isEqualToIgnoringWhitespace(expectedXml);
     }
 
+    @Test
+    public void When_MappingWithRelatedProblemWithIncorrectProblemExtensionExtensionUrl_Expect_UnspecifiedProblemWithOriginalText() {
```
Comment (martin-nhs) [inline service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterComponentsMapperTest.java:409]:
```suggestion
    public void When_MappingWithRelatedProblemWithIncorrectProblemExtensionExtensionUrl_Expect_UnspecifiedProblemWithOriginalText() {
```

When mapping what? 😄

Comment (martin-nhs) [inline service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterComponentsMapperTest.java:409]: When mapping what? 😄 (also can we make this package-private)

**Comment by martin-nhs** [comment]
Looks like End-to-End tests have failed:

`#18 134.1 curl: (28) Failed to connect to archive.apache.org port 443 after 133745 ms: Could not connect to server`

May need to manually re-run!

File: service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterComponentsMapperTest.java (hunk 1 above)
Comment (MartinWheelerMT) [reply service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterComponentsMapperTest.java:409]: When a related problem has an extension (with a correct problem extension url) which itself has an extension which has an incorrect URL!! (hard to fit into a test name with limited characters)
Made the added tests package private (even though it feels so wrong not having a access modifier)

**Comment by MartinWheelerMT** [comment]
> Looks like End-to-End tests have failed:
> 
> `#18 134.1 curl: (28) Failed to connect to archive.apache.org port 443 after 133745 ms: Could not connect to server`
> 
> May need to manually re-run!

Yeah, it should re-run now anyway, hopefully doesn't happen again!

File: service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterComponentsMapperTest.java (hunk 1 above)
Comment (martin-nhs) [reply service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterComponentsMapperTest.java]: I think we've discussed this on a call before, all examples provided by JUnit 5 tend to be package-private! I can see the benefit in them not being public (not totally against it), other packages shouldn't be concerned about the tests!

**Review by martin-nhs — APPROVED** [review]

**Pull request merged**